edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerModule.cs

```diff
@@ -84,25 +84,24 @@ public virtual bool Equals(IModule<DockerConfig> other)
                 return true;
             }
 
-            return string.Equals(this.Name, other.Name) &&
+            return this.BaseEquals(other) &&
                 string.Equals(this.Version, other.Version) &&
-                string.Equals(this.Type, other.Type) &&
-                this.DesiredStatus == other.DesiredStatus &&
-                this.Config.Equals(other.Config) &&
-                this.RestartPolicy == other.RestartPolicy &&
-                this.ImagePullPolicy == other.ImagePullPolicy &&
-                this.StartupOrder == other.StartupOrder &&
-                EnvDictionaryComparer.Equals(this.Env, other.Env);
+                this.Config.Equals(other.Config);
         }
 
         public virtual bool IsOnlyModuleStatusChanged(IModule other)
         {
             return other is DockerModule dockerModule &&
-                string.Equals(this.Name, other.Name) &&
+                this.BaseEquals(other) &&
                 string.Equals(this.Version, other.Version) &&
+                this.Config.Equals(dockerModule.Config);
+        }
+
+        protected bool BaseEquals(IModule other)
+        {
+            return string.Equals(this.Name, other.Name) &&
                 string.Equals(this.Type, other.Type) &&
                 this.DesiredStatus != other.DesiredStatus &&
```

Comment:
DesiredStatus should not be in BaseEquals method.  It is different between Equals and IsOnlyModuleStatusChanged methods.

Reply:
I think the same approach (extracting base hash code method) should use for GetHashCode as well.

Reply:
We should refactor and extract abstract base class for EdgeAgentDockerModule, EdgeHubDockerModule and DockerModule.  The base class should implements Equals (equivalent to what BaseEquals method) and GetHashCode method.